### PR TITLE
Add support for filtering report issues by severity/confidence/paths

### PR DESCRIPTION
**Pull request by zam0x1ma**
@ikkisoft This adds the Severity, Confidence and Path filters in the `/burp/report` endpoint, as suggested in #125

### CODE REVIEW

File: src/main/java/com/vmware/burp/extension/service/BurpService.java [hunk 1]
```diff
@@ -353,13 +357,43 @@ public void excludeFromScope(String urlString) throws MalformedURLException {
...
         Path reportFile = Files.createTempFile("Report", "." + reportType.getReportType());
         reportFile.toFile().deleteOnExit();
-        LegacyBurpExtender.getInstance().getCallbacks()
-                .generateScanReport(reportType.getReportType(),
-                        LegacyBurpExtender.getInstance().getCallbacks().getScanIssues(urlPrefix),
-                        reportFile.toFile());
+
+        IBurpExtenderCallbacks burpExtenderCallbacks = LegacyBurpExtender.getInstance().getCallbacks();
+
+        Supplier<Stream<IScanIssue>> scanIssuesSupplier = () -> Arrays.stream(burpExtenderCallbacks.getScanIssues(urlPrefix));
```
Comment (ikkisoft) [inline src/main/java/com/vmware/burp/extension/service/BurpService.java:367]: I am not sure it makes sense to have _urlPrefix_ as well as _issuePath_

_urlPrefix_  is actually a URL so in the form http(s)://doyensec.com/
_issuePath_ are just URI paths

We can leave as is, but I would adjust the description in the `ApiImplicitParam` to specify that for users

File: src/main/java/com/vmware/burp/extension/service/BurpService.java [hunk 2]
```diff
@@ -353,13 +357,43 @@ public void excludeFromScope(String urlString) throws MalformedURLException {
...
+        List<String> severities = Arrays.stream(issueSeverities)
+                .map(issueSeverity -> issueSeverity.getIssueSeverity())
+                .collect(Collectors.toList());
+        List<String> confidences = Arrays.stream(issueConfidences)
+                .map(issueConfidence -> issueConfidence.getIssueConfidence())
+                .collect(Collectors.toList());
+        Supplier<Stream<String>> pathsSupplier = () -> Arrays.stream(issuePaths)
+                .map(path -> path.startsWith("/") ? path : "/" + path);
+
+        IScanIssue[] filteredScanIssues = scanIssuesSupplier.get()
```
Comment (ikkisoft) [inline src/main/java/com/vmware/burp/extension/service/BurpService.java:382]: Very clean way of filtering issues

**Review by ikkisoft — CHANGES_REQUESTED** [review]
Besides the minor comment I made around paths and URL, i believe this is ready to be merged. 

Another improvement could be to write (or expand) the scan & report test in https://github.com/vmware/burp-rest-api/blob/master/src/test/java/com/vmware/burp/extension/client/BurpClientIT.java

**Comment by ikkisoft** [comment]
@zam0x1ma Code changes seem ok, but it's not breaking 

`com.vmware.burp.extension.client.BurpClientIT > testScannerSpiderAndReportMethods FAILED`

**Comment by zam0x1ma** [comment]
@ikkisoft I'm still working in the report test but I'm having issues to run the tests.
Even without my changes they all fail because Spring is not able to find Burp API classes.
Do you happen to know how to workaround this?

**Comment by zam0x1ma** [comment]
@ikkisoft Nevermind, I was able to run all the tests and fix the one failing.

**Comment by ikkisoft** [comment]
Reviewed all changes, and performed additional testing. All seems to be working great. Thanks @zam0x1ma for you contribution! Well done!

**Pull request merged**